recipes/battery-embed/all/conanfile.py

```diff
@@ -0,0 +1,73 @@
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.files import copy, get
+from conan.tools.cmake import cmake_layout
+from conan.tools.scm import Version
+import os
+
+required_conan_version = ">=1.53.0"
+
+class BatteryEmbedConan(ConanFile):
+    name = "battery-embed"
+    description = "A CMake/C++20 library to embed resource files at compile time"
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/batterycenter/embed"
+    topics = ("embed")
+    package_type = "build-scripts"
+    settings = "os", "arch", "compiler", "build_type"
+
+    @property
+    def _min_cppstd(self):
+        return 20
+
+    @property
+    def _compilers_minimum_version(self):
+        return {
+            "gcc": "11",
+            "clang": "12",
+            "apple-clang": "13",
+            "Visual Studio": "16",
+            "msvc": "192",
+        }
+
+    def export_sources(self):
+        copy(self, "embed.cmake", src=self.recipe_folder, dst=self.export_sources_folder)
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
+
+    def validate(self):
+        if self.settings.compiler.cppstd:
+            check_min_cppstd(self, self._min_cppstd)
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        if minimum_version and Version(self.settings.compiler.version) < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires C++{self._min_cppstd}, which your compiler does not support."
+            )
+
+    def package_id(self):
+        del self.info.settings.arch
+        del self.info.settings.compiler
+        del self.info.settings.build_type
```

Comment:
I'm confused. Why is this deleting every setting _but_ the os? As far as I can see, if the build() method only has 1 `copy()` call, where's the variance there depending on the os?

Reply:
You are right, there is no reason to have different packages per OS apparently
```suggestion
        del self.info.settings.build_type
        del self.info.settings.os
```
(tried removing the `settings` attribute but it's required for the `cmake_layout()`, so no luck)